### ISSUE TITLE
Unnecessary/incorrect error report (reportGeneralTypeIssues)

### ISSUE THREAD

**Issue by nevzatseferoglu** [microsoft/pyright#5403]
**Describe the bug**
Giving unnecessary error message.

**Expected behavior**
There should not be any error.

**Code or Screenshots**
![Screenshot 2023-06-29 at 03 57 31](https://github.com/microsoft/pyright/assets/43863606/4a186f22-0cca-4d6c-a3ae-ce2a39cd757d)

**Comment by nevzatseferoglu** [comment]
Also, I'm using `Neovim` and `pyright version 1.1.315`

**Comment by debonte** [comment]
This is by design. Pyright doesn't track the types of individual elements of lists. It sees the type of any element of a `list[Union[int | list[int]]]` as `Union[int | list[int]]`.

Since `__setitem__` and `append` are not supported on `int`, they are also not supported on `Union[int | list[int]]`.

One option is to test the type of the element before treating it as a `list`. For example:

```python
element = l[4]
if isinstance(element, list):
    element[0] = -1
    element.append(-3)
```

Depending on your scenario, switching to a tuple could also work:

```python
t = (1, 2, 3, 4, [5, 6])
t[4][0] = -1
t[4].append(-3)
```
